lib/foreman/renderer.rb

```diff
@@ -26,14 +28,42 @@ def template_logger
       @template_logger ||= Foreman::Logging.logger('templates')
     end
 
+    def host_enc(*path)
+      @enc ||= @host.info
+      return @enc.deep_dup if path.compact.empty?
+      enc = @enc
+      path.each { |step| enc = enc.fetch step }
+      enc
```

Comment:
No duplication here, but it does above if there is no path? This should probably dup too.